ctapipe/tools/dl1_merge.py

```diff
@@ -161,58 +170,79 @@ def setup(self):
         PROV.add_output_file(str(self.output_path))
 
         if self.skip_parameters is True and self.skip_images is True:
-            self.log.warning("Skip-parameters and skip-images are both" "set to True")
+            self.log.warning("Skip-parameters and skip-images are both set to True")
+            sys.exit(1)
 
         # Get input Files
         args = self.parser.parse_args(self.extra_args)
-        self.input_files = args.input_files
+        self.input_files.extend(args.input_files)
         if self.input_dir is not None:
             self.input_files.extend(sorted(self.input_dir.glob(self.file_pattern)))
+
         if not self.input_files:
-            self.input_files = sorted(Path(".").glob(self.file_pattern))
+            self.log.critical(
+                "No input files provided, either provide --input-dir "
+                "or input files as positional arguments"
+            )
+            sys.exit(1)
 
         # create output file with subarray from first file
         self.first_subarray = SubarrayDescription.from_hdf(self.input_files[0])
         self.first_subarray.to_hdf(self.output_path)
         self.output_file = tables.open_file(self.output_path, mode="a")
 
+        # setup required nodes
+        self.usable_nodes = set(all_nodes)
+
+        if self.skip_simu_images is True:
+            simu_images = ["/simulation/event/telescope/images"]
+            self.usable_nodes = self.usable_nodes - set(simu_images)
+
+        if self.skip_images is True:
+            images_path = [
+                "/simulation/event/telescope/images",
+                "/dl1/event/telescope/images",
+            ]
+            self.usable_nodes = self.usable_nodes - set(images_path)
+
+        if self.skip_parameters is True:
+            parameters_path = [
+                "/simulation/event/telescope/parameters",
+                "/dl1/event/telescope/parameters",
+            ]
+            self.usable_nodes = self.usable_nodes - set(parameters_path)
+
     def check_file(self, file):
         # Check that the file is not broken or any node is missing
         file_path = file.root._v_file.filename
         current_subarray = SubarrayDescription.from_hdf(file_path)
-        # Check subarrays
+
+        # Check subarray
         if self.first_subarray != current_subarray:
-            self.log.warning(f"Subarray does not match for {file_path}. " "Skip File.")
-            self.skip_file = True
+            self.log.critical(f"Subarray does not match for {file_path}")
+            self.broken_file = True
```

Comment:
Why not raise the error / log here? Instead of setting a member variable.

This member variable introduces some unneeded state I think.

Reply:
I thought it might be good to know if anything else is missing in that file, so that the whole check_file() could run through and raising all critical/warnings and in addition to that i would have just one if clause at the end (with 'self.skip_broken_files) to raise the error.

Reply:
You can still do that with a local variable. 

```
broken = False

if check_one_failed:
    broken = True

if check_two_failed:
   broken = True

return broken
```